store.go

```diff
@@ -152,12 +146,9 @@ func (n *network) getEndpointFromStore(eid string) (*endpoint, error) {
 	for _, store := range n.ctrlr.getStores() {
 		ep := &endpoint{id: eid, network: n}
 		err := store.GetObject(datastore.Key(ep.Key()...), ep)
-		if err != nil && err != datastore.ErrKeyNotFound {
-			return nil, fmt.Errorf("could not find endpoint %s: %v", eid, err)
-		}
-
 		// Continue searching in the next store if the key is not found in this store
-		if err == datastore.ErrKeyNotFound {
+		if err != nil {
+			log.Warnf("could not find endpoint %s: %v", eid, err)
```

Comment:
I think this one has a bug. `getEndpointFromStore` should only look at the store in which the network belongs rather than iterating through all the stores. Also we should just return an error here if endpoint is not found even if it is `KeyNotFound`